device-code-flow-console/MyInformation.cs

```diff
@@ -32,185 +31,46 @@
 
 namespace device_code_flow_console
 {
-    public class MyInformation
+    public class MyInformation : PublicAppUsingDeviceCodeFlow
     {
-        PublicClientApplication app;
+        public MyInformation(PublicClientApplication app, HttpClient client) : base(app, client)
+        {
+        }
 
         /// <summary>
         /// Scopes to request access to the protected Web API (here Microsoft Graph)
         /// </summary>
-        public static string[] Scopes { get; set; } = new string[] { "user.read", "User.ReadBasic.All" };
+        private static string[] Scopes { get; set; } = new string[] { "User.Read", "User.ReadBasic.All"};
 
         /// <summary>
-        /// URLs of the protected Web APIs to call (here Microsoft Graph)
+        /// URLs of the protected Web APIs to call (here Microsoft Graph endpoints)
         /// </summary>
-        public static string WebApiUrlMe { get; set; } = "https://graph.microsoft.com/v1.0/me";
-        public static string WebApiUrlMyManager { get; set; } = "https://graph.microsoft.com/v1.0/me/manager";
+        private static string WebApiUrlMe { get; set; } = "https://graph.microsoft.com/v1.0/me";
+        private static string WebApiUrlMyManager { get; set; } = "https://graph.microsoft.com/v1.0/me/manager";
 
         /// <summary>
         /// Calls the Web API and displays its information
         /// </summary>
         /// <returns></returns>
         public async Task DisplayMeAndMyManagerAsync()
         {
-            AuthenticationConfig config = AuthenticationConfig.ReadFromJsonFile("appsettings.json");
-
-            app = new PublicClientApplication(config.ClientId, config.Authority);
-
-            AuthenticationResult authenticationResult = await AcquireATokenFromCacheOrDeviceCodeFlow();
+            AuthenticationResult authenticationResult = await AcquireATokenFromCacheOrDeviceCodeFlow(Scopes);
```

Comment:
This method does too much, please create a private method for displaying the graph stuff, something like CallGraphAndDisplayResults(string authToken)

Reply:
done. Thanks!